aif360/algorithms/intersectional_fairness.py

```diff
@@ -0,0 +1,881 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright 2023 Fujitsu Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+import math
+import collections as cl
+import traceback
+import concurrent.futures
+
+from aif360.datasets import StructuredDataset
+from aif360.datasets import BinaryLabelDataset
+from aif360.metrics import BinaryLabelDatasetMetric
+from aif360.metrics import ClassificationMetric
+
+from aif360.algorithms.isf_helpers.isf_utils import const
+from aif360.algorithms.isf_helpers.isf_utils.common import create_multi_group_label
+from aif360.algorithms.isf_helpers.preprocessing.preprocessing import PreProcessing
+from aif360.algorithms.isf_helpers.inprocessing.inprocessing import InProcessing
+from aif360.algorithms.isf_helpers.postprocessing.postprocessing import PostProcessing
+
+from aif360.algorithms.isf_helpers.preprocessing.massaging import Massaging
+from aif360.algorithms.isf_helpers.inprocessing.adversarial_debiasing import AdversarialDebiasing
+from aif360.algorithms.isf_helpers.postprocessing.reject_option_based_classification import RejectOptionClassification
+
+from tqdm import tqdm
+
+class IntersectionalFairness():
```

Comment:
can we have this extend `aif360.algorithms.Transformer`?
```suggestion
class IntersectionalFairness(Transformer):
```

Reply:
indeed, thank you, we have now extended as suggested